imgee/views/files.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+from flask import (flash, g, jsonify, redirect, render_template, request, url_for)
+from sqlalchemy import and_
+
+from coaster.views import load_model, load_models
+from imgee import app, lastuser
+from imgee.forms import (
+    UploadImageForm, UpdateTitle, EditTitleForm,
+    AddLabelForm, DeleteImageForm
+)
+from imgee.models import StoredFile, db, Profile
+from imgee.storage import delete, save_file
+from imgee.utils import get_media_domain
+from .index import get_prev_next_images
+import imgee.utils as utils
+
+
+def _redirect_url_frm_upload(profile_name):
+    # if the referrer is from 'pop_up_gallery' redirect back to referrer.
+    referrer = request.referrer or ''
+    if url_for('pop_up_gallery', profile=profile_name) in referrer:
+        url = request.referrer
+    else:
+        url = url_for('profile_view', profile=profile_name)
+    return url
+
+
+@app.route('/<profile>/new', methods=['GET', 'POST'])
+@lastuser.requires_login
+@load_model(Profile, {'name': 'profile'}, 'profile',
+    permission=['new-file', 'siteadmin'])
+def upload_file(profile):
+    upload_form = UploadImageForm()
+    if upload_form.validate_on_submit():
+        file_ = request.files['file']
+        title, stored_file = save_file(file_, profile=profile)
+        flash('"%s" uploaded successfully.' % title)
+        return redirect(_redirect_url_frm_upload(profile.name))
+    return render_template('form.html', form=upload_form, profile=profile)
+
+
+@app.route('/<profile>/new.json', methods=['POST'])
+@lastuser.requires_login
+@load_model(Profile, {'name': 'profile'}, 'profile',
+    permission=['new-file', 'siteadmin'])
+def upload_file_json(profile):
```

Comment:
This method could be handled with a `coaster.views.render_with` wrapper on the above method. File a separate ticket for this?